R/metrics-ordinal.R

```diff
@@ -0,0 +1,160 @@
+#'   probabilities in `predicted` correspond to.
+#' @returns A numeric vector of size n with log scores
+#' @inheritSection illustration-input-metric-ordinal Input format
+#' @importFrom methods hasArg
+#' @export
+#' @keywords metric
+#' @family log score functions
+#' @examples
+#' factor_levels <- c("one", "two", "three")
+#' predicted_label <- factor(c("one", "two", "three"), levels = factor_levels)
+#' observed <- factor(c("one", "three", "two"), levels = factor_levels)
+#' predicted <- matrix(c(0.8, 0.1, 0.4,
+#'                       0.1, 0.2, 0.4,
+#'                       0.1, 0.7, 0.2),
+#'                     nrow = 3)
+#' logs_nominal(observed, predicted, predicted_label)
+logs_ordinal <- function(observed, predicted, predicted_label) {
+  assert_input_ordinal(observed, predicted, predicted_label)
+  n <- length(observed)
+  if (n == 1) {
+    predicted <- matrix(predicted, nrow = 1)
+  }
+  observed_indices <- as.numeric(observed)
+  pred_for_observed <- predicted[cbind(1:n, observed_indices)]
+  logs <- -log(pred_for_observed)
+  return(logs)
+}
+
+#' Ranked Probability Score for ordinal outcomes
+#'
+#' @description
+#' The Ranked Probability Score (RPS) measures the difference between the predicted
+#' and observed cumulative distribution functions. It is a proper scoring rule that
+#' takes the ordering of categories into account. Small values are better
+#' (best is zero, worst is 1).
+#' @param observed A factor of length n with N levels holding the observed
+#'   values.
+#' @param predicted nxN matrix of predictive probabilities, n (number of rows)
+#'   being the number of observations and N (number of columns) the number of
+#'   possible outcomes.
+#' @param predicted_label A factor of length N, denoting the outcome that the
+#'   probabilities in `predicted` correspond to.
+#' @returns A numeric vector of size n with ranked probability scores
+#' @inheritSection illustration-input-metric-nominal Input format
+#' @importFrom methods hasArg
+#' @importFrom scoringRules rps_probs
+#' @export
+#' @keywords metric
+#' @family scoring functions
+#' @examples
+#' factor_levels <- c("one", "three", "two")
+#' predicted_label <- factor(factor_levels, levels = factor_levels, ordered = TRUE)
+#' observed <- factor(c("three", "three", "two"), levels = factor_levels, ordered = TRUE)
+#' predicted <- matrix(
+#'   c(0.8, 0.1, 0.4,
+#'     0.1, 0.2, 0.4,
+#'     0.1, 0.7, 0.2),
+#'   nrow = 3
+#' )
+#' rps_ordinal(observed, predicted, predicted_label)
+rps_ordinal <- function(observed, predicted, predicted_label) {
+  assert_input_ordinal(observed, predicted, predicted_label)
+  n <- length(observed)
+  if (n == 1) {
+    predicted <- matrix(predicted, nrow = 1)
+  }
+
+  # Reorder the predicted matrix columns to match the natural ordering
+  correct_order <- as.numeric(predicted_label)
+  ordered_predicted <- predicted[, correct_order]
+
+  # Use scoringRules implementation
```

Comment:
```suggestion
```

This comment is followed exactly the same thing in very simple code suggest remove